HeterogeneousCore/SonicCore/src/SonicClientBase.cc

```diff
@@ -40,22 +55,32 @@ void SonicClientBase::start(edm::WaitingTaskWithArenaHolder holder) {
   holder_ = std::move(holder);
 }
 
-void SonicClientBase::start() { tries_ = 0; }
+void SonicClientBase::start() {
+  totalTries_ = 0;
+  // initialize all actions
+  for (const auto& action : retryActions_) {
+    action->start();
+  }
+}
 
 void SonicClientBase::finish(bool success, std::exception_ptr eptr) {
   //retries are only allowed if no exception was raised
   if (!success and !eptr) {
-    ++tries_;
-    //if max retries has not been exceeded, call evaluate again
-    if (tries_ < allowedTries_) {
-      evaluate();
-      //avoid calling doneWaiting() twice
-      return;
+    ++totalTries_;
+    // Check if any retry actions are still valid
+    bool anyRetryAllowed = false;
```

Comment:
this variable is essentially unused? initialized to false and never touched again; only used to issue the exception below. since this loop returns if an action triggers, it seems like `if (!anyRetryAllowed)` below could just be omitted, and then this variable would not be needed at all...

Reply:
Agree and removed.